### DISCUSSION TITLE
endWrite() and startWrite()

### DISCUSSION THREAD

**Discussion by Stredoslovak** [Bodmer/TFT_eSPI#1955] [Q&A - General]
Hello, I'm in a kind of a pitch. I'm running out of pins on my ESP32 and I need to add an SD card functionality to the project.
**PROBLEM**
I am using a 1.8 SPI TFT display (ST7735) connected to the SPI bus of the ESP32 
I want to connect to the same pins with SD card and read/write the SD, in order to do that i need to end all communication related to the display (cuz that's how spi works)
**QUESTION**
From the documented pieces of code, what I understood was that: 
If I will use the endWrite() function to pull the CS pin high I should be able to then communicate with the SD card (provided that I also pull the card CS pin low)
Now, when I want to resume communication with the display, I should use the startWrite() function. 
Problem being that startWrite() has no comment at all and I don't know what will it do, as of now I don't have any test rig built so I can't test this myself. In theory communicating with other devices on the same bus is 100% possible, but given how do some libraries handle SPI communication it might turn out a problem.

_Now, is this right or are there any other steps to use the same SPI bus to communicate with other devices?_

**Comment by Frank-Bemelman** [comment]
Here are some thoughts of my own.. 

https://github.com/Bodmer/TFT_eSPI/discussions/1885

**Reply by Stredoslovak** [reply]
Thank you very muchbfor your answer.
Okay, I've read trough the post you've mentioned.
Pretty much just confirmed what I've already knew.
But still could you please provide any inside into the startWrite() and endWrite() functions?

**Reply by Frank-Bemelman** [reply]
In TFT_eSPI.cpp you can see what it does, but I think you don't have to worry about it. If you perform some action on the display, these calls are made by the library already. And CS is high once a graphic operation has completed. 

I guess you can have problems if you have asynchronous tasks doing display updates and SD card acces. You would have to insert those waiting for CS to rise before you can start a TFT action or SD action. As described in my thoughts on using both. If your program is just one task and everything inside loop() there is little to worry about - calls to the TFT library and calls to SD library are in sequence, not quasi simultaneously. 

Not sure what you are worried about ;-)
